File: java-vfs/src/main/java/emg/java/vfs/commands/Touch.java
package emg.java.vfs.commands;

import emg.java.vfs.files.DirEntry;
import emg.java.vfs.files.File;
import emg.java.vfs.filesystem.State;

public class Touch extends CreateEntry {
    public Touch(String name) {
        super(name);
    }

    @Override
    public DirEntry createSpecificEntry(State state, String entryName) {
        return File.empty(state.wd().path(), entryName);
    }
}

File: scala-vfs/src/main/scala/emg/scala/vfs/commands/Touch.scala
package emg.scala.vfs.commands

import emg.scala.vfs.files.{DirEntry, File}
import emg.scala.vfs.filesystem.State

class Touch(name: String) extends CreateEntry(name) {
  override def createSpecificEntry(state: State, entryName: String): DirEntry = File.empty(state.wd.path, name)
}
